protomap/proto.go

```diff
@@ -177,13 +177,100 @@ func parseList(fd protoreflect.FieldDescriptor, v protoreflect.Value, vals map[*
 
 	// TODO(robjs): This handles the case where we have a list - but not a leaf-list.
 	//              Add implementation to handle leaf lists.
+
+	leaflist, leaflistunion, err := annotatedYANGFieldInfo(fd)
+	if err != nil {
+		return fmt.Errorf("cannot parse list field %s, %v", fd.FullName(), err)
+	}
+
 	l := v.List()
+
+	if leaflist {
```

Comment:
Use a switch statement to separate out the three cases? This function is pretty long -- maybe extract each case into its own helper function?

Reply:
Makes sense to me. Done.